src/components/SignUp/components/signup-interest-selection/SignupInterestSelection.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useContext, useState } from "react";
+import { interests } from "../../constants";
+import SelectBox from "../../../select-box/SelectBox";
+import InfoBox from "../../../info-box/InfoBox";
+import { AppContext } from "../../../../context/AppContext";
+
+const CINEMA = "Sinema";
+const RESTAURANT = "Restoran ve yemek";
+const CONCERT = "Konserler ve canlı müzik";
+const THEATRE = "Tiyatro";
+const ART_GALLERY = "Sanat sergileri ve galeriler";
+const SPORTS = "Spor";
+const OUTDOOR = "Doğa yürüyüşleri ve açık hava";
+const CLUBS = "Gece kulüpleri ve eğlence mekanları";
+const READING = "Kitap okuma ve edebiyat";
+const MUSEUM = "Müzeler";
+const KIDS_ACTIVITIES = "Çocuk etkinlikleri";
+const YOGA = "Yoga ve spor salonları";
+
+function SignupInterestSelection() {
+  const [selectedInterests, setSelectedInterests] = useState([]);
+  const { state } = useContext(AppContext);
+
+  function changeInterestsToEnum(interests) {
+    let interestsEnum = [];
+    interests.forEach((interest) => {
+      switch (interest.name) {
+        case CINEMA:
+          interestsEnum.push("CINEMA");
+          break;
+        case RESTAURANT:
+          interestsEnum.push("RESTAURANT");
+          break;
+        case CONCERT:
+          interestsEnum.push("CONCERT");
+          break;
+        case THEATRE:
+          interestsEnum.push("THEATRE");
+          break;
+        case ART_GALLERY:
+          interestsEnum.push("ART_GALLERY");
+          break;
+        case SPORTS:
+          interestsEnum.push("SPORTS");
+          break;
+        case OUTDOOR:
+          interestsEnum.push("OUTDOOR");
+          break;
+        case CLUBS:
+          interestsEnum.push("CLUBS");
+          break;
+        case READING:
+          interestsEnum.push("READING");
+          break;
+        case MUSEUM:
+          interestsEnum.push("MUSEUM");
+          break;
+        case KIDS_ACTIVITIES:
+          interestsEnum.push("KIDS_ACTIVITIES");
+          break;
+        case YOGA:
+          interestsEnum.push("YOGA");
+          break;
+        default:
+          break;
+      }
+    });
+    return interestsEnum;
+  }
+
+  function handleSetInterestsForUser() {
+    // request body will look like this: ["MUSEUM","YOGA","READING","OUTDOOR"] (string)
```

Comment:
@mukrime-s @ThePoetDev 
Can you guys take a look at the usage of **POST** request to`http://localhost:8080/user/<user-id>/interests`. _(copilot recommended this structure, it sounds clean_ 🤔 _)_

The request body will be `["MUSEUM","YOGA","READING","OUTDOOR"] string`. Any feedbacks are welcomed 🙌🏼

Reply:
We have a user-interest table, so we store it there, and it doesn't seem logical to me to keep interest information in the user data. There should be a many-to-many relationship between user and interest.

Reply:
Oh I see. Then we'll do it this way 👍🏼

Reply:
Good luck, good work  🙌🏼